src/otx/v2/adapters/torch/mmengine/mmaction/modules/datasets/pipelines/loading.py

```diff
@@ -0,0 +1,49 @@
+"""Collection of video loading data pipelines for OTX Action Task."""
+
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+from mmaction.registry import TRANSFORMS
+
+if TYPE_CHECKING:
+    from datumaro.components.dataset import Dataset as DatumDataset
+
+@TRANSFORMS.register_module(force=True)
+class OTXRawFrameDecode:
+    """Load and decode frames with given indices."""
+
+    otx_dataset: DatumDataset
+
+    def __call__(self, results: dict[str, Any]) -> dict[str, Any]:
+        """Call function of RawFrameDecode."""
+        return self._decode_from_list(results)
+
+    def _decode_from_list(self, results: dict[str, Any]) -> dict:
+        """Generate numpy array list from list of DatasetItemEntity."""
+        imgs = []
+        for index in results["frame_inds"]:
+            item_id = results["item_ids"][index]
+            imgs.append(self.otx_dataset.get(id=item_id, subset="annotations").media.data)
+
+        results["imgs"] = imgs
+        results["original_shape"] = imgs[0].shape[:2]
+        results["img_shape"] = imgs[0].shape[:2]
+
+        # we resize the gt_bboxes and proposals to their real scale
+        if "gt_bboxes" in results:
+            height, width = results["img_shape"]
+            scale_factor = np.array([width, height, width, height])
+            gt_bboxes = results["gt_bboxes"]
+            gt_bboxes = (gt_bboxes * scale_factor).astype(np.float32)
```

Comment:
Do we have ActionDetectionDataset? This scaling might be affected by this.

Reply:
Action detection has not been considered yet. I removed this part. https://github.com/openvinotoolkit/training_extensions/pull/2627/commits/fd28202eba744fe7cacef5fd3226a658b71159b3